lib/govuk_content_models/presentation_toggles.rb

```diff
@@ -0,0 +1,39 @@
+module PresentationToggles
+  extend ActiveSupport::Concern
+
+  included do
+    field :presentation_toggles, type: Hash, default: default_presentation_toggles
+    validates_presence_of :organ_donor_registration_promotion_url, if: :display_organ_donor_registration_promotion?
+  end
+
+  def display_organ_donor_registration_promotion=(value)
+    value = value.is_a?(Boolean) ? value : value != '0' # if assigned using a checkbox
+    organ_donor_registration_promotion_key['display'] = value
+  end
+
+  def display_organ_donor_registration_promotion
+    organ_donor_registration_promotion_key['display']
+  end
+  alias_method :display_organ_donor_registration_promotion?, :display_organ_donor_registration_promotion
+
+  def organ_donor_registration_promotion_url=(value)
+    organ_donor_registration_promotion_key['url'] = value
+  end
+
+  def organ_donor_registration_promotion_url
+    return "" unless display_organ_donor_registration_promotion?
+    organ_donor_registration_promotion_key['url']
+  end
+
+  def organ_donor_registration_promotion_key
+    presentation_toggles['organ_donor_registration_promotion']
+  end
+
+  module ClassMethods
+    def default_presentation_toggles
+      {
+        'organ_donor_registration_promotion' => { 'url' => '', 'display' => false }
+      }
+    end
+  end
+end
```

Comment:
I think I can see your point. We could serve the goal of flexibility without needing to use metaprogramming by using an embedded set of documents. I've sketched it out here: https://github.com/alphagov/govuk_content_models/compare/edition-has-display-toggles_embed?expand=1 There are some niggles I haven't solved with doing it that way - eg how do you ensure uniqueness of toggles in the set?